### ISSUE TITLE
convert a Mermaid graph in obsidian to Excalidraw editable graph or support mermaid syntax in Excalidraw

### ISSUE THREAD

**Issue by Ravenclawer** [zsviczian/obsidian-excalidraw-plugin#116]
**Is your feature request related to a problem? Please describe.**
I was using Mermaid in Obsidian and it’s quite convenient to draw a flowchart since you don’t have to worry about the position and the size of any squares, elliptic or triangles, etc.
Excalidraw elevates drawing and note organization to another level. The elements in the drawing could actually link to notes and texts in the obsidian vault, which Mermaid could not possibly realize.
**Describe the solution you'd like**
So, I wonder if you could combine mermaid with Excalidraw or add Mermaid-like graph making syntax to Excalidraw to combine the intuition of Excalidraw and the ### convenience of Mermaid
**Describe alternatives you've considered**

**Additional context**

**Comment by Ravenclawer** [comment]
I found mermaid could link to files by the following syntax:
```mermaid
graph LR
A[Note Name] --> B 

class A internal-link;
```


https://forum.obsidian.md/t/hassle-free-linking-in-mermaid-graphs/6983/8#:~:text=graph%20LR%0AA%5BNote%20Name%5D%20--%3E%20B%20%0A%0Aclass%20A%20internal-link%3B

**Comment by zsviczian** [comment]
I agree this would be great! However, this would require a significant amount of development. I may get to this sometime in the future, but not anytime in the near term...

**Comment by jzhang-brex** [comment]
+1 to this. The use case is that we can use mermaid to quickly put in some draft and make it more free form in Excalidraw.

Question: If this feature is made, is it specific to the Obsidian Excalidraw or could we use it with Excalidraw in general (e.g. https://excalidraw.com/)

**Comment by AMGMNPLK** [comment]
+1

Is there any other way to vote for this feature than comment?

**Comment by kishorekumarxyz** [comment]
+1

**Comment by NOOS3NCE** [comment]
+1

**Comment by zsviczian** [comment]
Reverse engineer how it is implemented for visual studio code: https://marketplace.visualstudio.com/items?itemName=nopeslide.vscode-drawio-plugin-mermaid

**Comment by dfenerski** [comment]
+1, would be nice, sorry for the spam

**Comment by zsviczian** [comment]
This is coming this weekend :) an early beta is already available via BRAT.

**Comment by wgandhi** [comment]
I use the import mermaid feature in the plugin. However, when I use the sequenceDiagram, the column names appear in the wrong order.
I believe the issue is with the indentation in my mermaid syntax but cannot figure out what indentation I need to fix the picture. I have been using 4 spaces but that does not work consistently

**Comment by zsviczian** [comment]
Yeah, I believe this has been fixed on Excalidraw.com... but the issue is Obsidian has a different version of mermaid.js and mermaid is notorious for compatibility issues between versions. I am using the Obsidian built-in mermaid package to save on memory and limit the plugin size, but as a consequence, I haven't been able to deploy fixes to Obsidian. For now, it is what it is. I do not plan to burn time on trying to figure out mermaid and fix these issues - sorry.

**Comment by mlao-pdx** [comment]
Hi Zsolt! First, thank you for an incredible plugin. I'm currently vibe coding DataviewJS to generate dynamic narrative dashboards, and I'm hitting the limits of Obsidian's native Mermaid renderer (especially regarding theme-safe SVG exports and click interaction limits). While ExcalidrawAutomate is perfect for dynamic vector generation, manually calculating X/Y coordinates for auto-flowing charts (ea.addRect, etc., for nested subgraphs) in a background script is incredibly prohibitive compared to the Sugiyama math engine Mermaid uses.

So, what I have been doing is generating Mermaid syntax from my Vault using dataviewjs and then pasting it into the mermaid to Excalidraw functionality. Then I got to wondering...

Since the plugin already bundles the upstream @excalidraw/mermaid-to-excalidraw package for the UI modal, would it be possible to expose a headless method in the EA API (e.g., ea.addMermaid(mermaidString))? This would allow background scripts to feed raw Mermaid syntax directly to EA, letting Excalidraw silently compute the complex auto-layout and instantly output native, theme-aware Excalidraw elements to the canvas or an inline SVG. Exposing this headless parser would perfectly bridge the gap between "living" Markdown data queries (like Dataview) and Excalidraw's visual canvas natively, without requiring manual copy-pasting through the UI importer.

This seems like it might be a good midpoint between fully integrating mermaid into your plugin and the current state of mermaid in native Obsidian.

Thanks for considering this.

**Comment by zsviczian** [comment]
@mlao-pdx this is all already fully possible. ExcalidrawAutomate already exposes addMermaid, and you can create any Excalidraw file, or even PNG or SVG fully silently from code, even via Obsidian CLI.

**Comment by zsviczian** [comment]
This feature was implemented years ago. I am closing this issue.
